legal-api/src/legal_api/models/db.py

```diff
@@ -11,17 +11,222 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Create SQLAlchenmy and Schema managers.
+"""Create SQLAlchemy and Schema managers.
 
 These will get initialized by the application using the models
 """
+
 from flask_sqlalchemy import SQLAlchemy
-from sqlalchemy_continuum import make_versioned
+from sqlalchemy.ext.declarative import declarative_base
+from sql_versioning import TransactionManager as NewTransactionManager
+from sqlalchemy.orm import sessionmaker, configure_mappers
+from sqlalchemy import event, Column, DateTime, func, Integer
+from sqlalchemy_continuum import make_versioned, versioning_manager
+from sqlalchemy.orm.session import Session as SQLAlchemySession
+
+Base = declarative_base()
 
+class Transaction(Base):
+    __tablename__ = 'transaction'
 
-# by convention in the Flask community these are lower case,
-# whereas pylint wants them upper case
-db = SQLAlchemy()  # pylint: disable=invalid-name
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    issued_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
 
-# make_versioned(user_cls=None, plugins=[FlaskPlugin()])
+    def __init__(self, manager=None):
+        pass  # We don't need to do anything with the manager in this simplified version
+
+# Initialize SQLAlchemy-Continuum at the module level
 make_versioned(user_cls=None)
+
+def initialize_sqlalchemy_continuum(app):
+    with app.app_context():
+        versioning_manager.declarative_base = db.Model
+        versioning_manager.transaction_cls = Transaction
+        versioning_manager.options['native_versioning'] = False
+        configure_mappers()
+        print("SQLAlchemy-Continuum initialized")
+
+class VersioningSwitchableSession(SQLAlchemySession):
```

Comment:
custom db session that uses correct library for various functionality based off of db versioning enabled